ui/src/app/shared/components/application-selector/application-selector.tsx

```diff
@@ -6,6 +6,30 @@ import {appInstanceName, appQualifiedName, ComparisonStatusIcon, HealthStatusIco
 import {AuthSettingsCtx} from '../../context';
 
 export const ApplicationSelector = ({apps, formApi}: {apps: models.Application[]; formApi: FormFunctionProps}) => {
+    let reorderedApps: models.Application[] = [];
+    const appOfAppsArray = apps.filter(app => app.isAppOfAppsPattern);
```

Comment:
Same problem here. Let's follow the same pattern as above to reduce the amount of times we loop through the apps list.